addons/ofxOsc/src/ofxOscBundle.h

```diff
@@ -1,64 +1,50 @@
- 
- THIS SOFTWARE IS PROVIDED BY DAMIAN STEWART ''AS IS'' AND ANY
- EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
- WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
- DISCLAIMED. IN NO EVENT SHALL DAMIAN STEWART BE LIABLE FOR ANY
- DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
- (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
- LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
- ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
- (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
- SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
- */
-
+// copyright (c) openFrameworks team 2010-2017
+// copyright (c) Damian Stewart 2007-2009
 #pragma once
 
-#include <vector>
 #include "ofxOscMessage.h"
 
-class ofxOscBundle
-{
-public:	
+/// \class ofxOscBundle
+/// \brief an OSC bundle of ofxMessages and/or other ofxOscBundles
+class ofxOscBundle{
+public:
+
 	ofxOscBundle();
 	~ofxOscBundle();
-	ofxOscBundle( const ofxOscBundle& other ) { copy ( other ); }
-	ofxOscBundle& operator= ( const ofxOscBundle& other ) { return copy( other ); }
+	ofxOscBundle(const ofxOscBundle& other);
+	ofxOscBundle& operator=(const ofxOscBundle& other);
 	/// for operator= and copy constructor
-	ofxOscBundle& copy( const ofxOscBundle& other );
+	ofxOscBundle& copy(const ofxOscBundle& other);
 	
-	/// erase contents
-	void clear() { messages.clear(); bundles.clear(); }
+	/// clear bundle & message contents
+	void clear();
 
-	/// add bundle elements
-	void addBundle( const ofxOscBundle& element );
-	void addMessage( const ofxOscMessage& message );
+	/// add another bundle to the bundle
+	void addBundle(const ofxOscBundle& element);
+	
+	/// add a message to the bundle
+	void addMessage(const ofxOscMessage& message);
+	
+	/// \return the current bundle count
+	int getBundleCount() const;
+	
+	/// \return the current message count
+	int getMessageCount() const;
+	
+	/// \return the bundle at the given index
+	const ofxOscBundle& getBundleAt(int i) const;
+	
+	/// \return the bundle at the given index
+	ofxOscBundle& getBundleAt(int i);
+	
+	/// \return the message at the given index
+	const ofxOscMessage& getMessageAt(int i) const;
 	
-	/// get bundle elements
-	int getBundleCount() const { return bundles.size(); }
-	int getMessageCount() const { return messages.size(); }
-	/// return the bundle or message at the given index
-	const ofxOscBundle& getBundleAt( int i ) const { return bundles[i]; }
-    ofxOscBundle& getBundleAt( int i ) { return bundles[i]; }
-	const ofxOscMessage& getMessageAt( int i ) const { return messages[i]; }
-    ofxOscMessage& getMessageAt( int i ) { return messages[i]; }
+	/// \return the message at the given index
+	ofxOscMessage& getMessageAt(int i);
 	
 private:
 		
-	std::vector< ofxOscMessage > messages;
-	std::vector< ofxOscBundle > bundles;
```

Comment:
we are moving towards using std:: explicitly in headers to at some point remove the global include for std so better to leave these as they were